tests/SequenceTests.test.ts

```diff
@@ -0,0 +1,165 @@
+    userWithSeq.sharedSequencesReadAccess = [];
+    userWithSeq.sharedSequencesWriteAccess = [];
+    await userWithSeq.save();
+
+    const userToShareWith = new DBUser();
+    userToShareWith.mail = 'anatoli@karpov.de';
+    userToShareWith.id = 2351951;
+    userToShareWith.type = UserClass.REGULAR;
+    userToShareWith.password = await bcrypt.hash('verySecurePW', 13);
+    userToShareWith.name = 'anatoli';
+    userToShareWith.sharedSequencesReadAccess = ['CODE66'];
+    userToShareWith.sharedSequencesWriteAccess = ['CODE66'];
+    await userToShareWith.save();
+
+    const testSlide = new DBSlide();
+    testSlide.layout = LayoutType.TITLEPAGE;
+    testSlide.content = {};
+    testSlide.backgroundColor = 'white';
+    testSlide.sequenceCode = 'CODE66';
+    testSlide.order = 1;
+    testSlide.id = 10404;
+    await testSlide.save();
+});
+afterAll(async () => {
+    await mockDb.destroy();
+});
+describe('SequenceController Tests', () => {
+    //createNewSequence function
+    it('create sequence correctly', async () => {
+        await SequenceController.createNewSequence('newSequence', 931943);
+        const sequencesByUser = await DBSequence.findBy({ authorId: 931943 });
+        expect(sequencesByUser.length).toEqual(2);
+    });
+
+    //getSequenceByCode function
+    it('get existing sequence by ID', async () => {
+        const sequenceToGet = await DBSequence.findBy({ name: 'Sequence66' });
+        const sequence = await SequenceController.getSequenceByCode(
+            sequenceToGet[0].code
+        );
+        expect(sequence).toBeInstanceOf(DBSequence);
+    });
+    //getSequenceWithSlides function
+    it('returns existing sequence and its slides', async () => {
+        const testSlide = await DBSlide.findBy({ sequenceCode: 'CODE66' });
+        const seqWithSlides = await SequenceController.getSequenceWithSlides(
+            'CODE66'
+        );
+        expect(seqWithSlides.code).toEqual('CODE66');
+        expect(seqWithSlides.slides.length).toEqual(1);
+        expect(seqWithSlides.slides[0]).toEqual(testSlide[0]);
+    });
+
+    //saveSequence function
+    it('returns existing sequence and its slides with changes', async () => {
+        const sequenceChanges: Partial<ISequenceWithSlides> = {
+            name: 'changed',
+            code: 'CODE66',
+        };
+        await SequenceController.saveSequence(sequenceChanges);
+        const sequenceChanged = await DBSequence.findBy({ code: 'CODE66' });
+        expect(sequenceChanged[0].name).toEqual('changed');
+    });
+
+    //getSequencesOfUser function
+    it('try to get sequences from user', async () => {
+        const sequences = await SequenceController.getSequencesOfUser(931943);
+        expect(sequences.length).toEqual(2);
+        expect(sequences[0].code).toEqual('CODE66');
+    });
+
+    /**
+     * getSharedSequencesOfUser function
+     * NOTE: If you have read and write access set, then the sequence is returned twice. Intended?
```

Comment:
Not intended i guess. since we wont fix it in this phase, i suggest opening an issue, to keep an eye on it

Reply:
Not necessarily intended, but not "unintended" either, a user never has both read and write access, since write access includes read access, so they should never be part of both arrays